node/src/blockchain/blockchain_interface_utils.rs

```diff
@@ -0,0 +1,1412 @@
+        ));
+    }
+
+    #[test]
+    fn sign_transaction_just_works() {
+        let port = find_free_port();
+        let (_event_loop_handle, transport) = Http::with_max_parallel(
+            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
+            REQUESTS_IN_PARALLEL,
+        )
+        .unwrap();
+        let web3 = Web3::new(transport.clone());
+        let chain = DEFAULT_CHAIN;
+        let amount = 11_222_333_444;
+        let gas_price_in_gwei = 123000000000_u64;
+        let nonce = U256::from(5);
+        let recipient_wallet = make_wallet("recipient_wallet");
+        let consuming_wallet = make_paying_wallet(b"consuming_wallet");
+        let consuming_wallet_secret_key = consuming_wallet.prepare_secp256k1_secret().unwrap();
+        let data = sign_transaction_data(amount, recipient_wallet.clone());
+        let tx_parameters = TransactionParameters {
+            nonce: Some(nonce),
+            to: Some(chain.rec().contract),
+            gas: gas_limit(data, chain),
+            gas_price: Some(to_wei(gas_price_in_gwei)),
+            value: U256::zero(),
+            data: Bytes(data.to_vec()),
+            chain_id: Some(chain.rec().num_chain_id),
+        };
+        let result = sign_transaction(
+            chain,
+            Web3::new(Batch::new(transport)),
+            recipient_wallet,
+            consuming_wallet,
+            amount,
+            nonce,
+            gas_price_in_gwei,
+        );
+
+        let expected_tx_result = web3
+            .accounts()
+            .sign_transaction(tx_parameters, &consuming_wallet_secret_key)
+            .wait()
+            .unwrap();
+
+        assert_eq!(result, expected_tx_result);
+    }
+
+    #[test]
+    #[should_panic(
+        expected = "Consuming wallet doesnt contain a secret key: Signature(\"Cannot sign with non-keypair wallet: Address(0x00000000636f6e73756d696e675f77616c6c6574).\")"
+    )]
+    fn sign_transaction_panics_on_bad_consuming_wallet() {
+        let port = find_free_port();
+        let (_event_loop_handle, transport) = Http::with_max_parallel(
+            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
+            REQUESTS_IN_PARALLEL,
+        )
+        .unwrap();
+        let chain = DEFAULT_CHAIN;
+        let amount = 11_222_333_444;
+        let gas_price_in_gwei = 123000000000_u64;
+        let nonce = U256::from(5);
+        let recipient_wallet = make_wallet("recipient_wallet");
+        let consuming_wallet = make_wallet("consuming_wallet");
+
+        let _result = sign_transaction(
+            chain,
+            Web3::new(Batch::new(transport)),
+            recipient_wallet,
+            consuming_wallet,
+            amount,
+            nonce,
+            gas_price_in_gwei,
+        );
+    }
+
+    #[test]
+    #[should_panic(expected = "Signing should be done locally")]
+    fn sign_transaction_locally_panics_on_signed_transaction() {
```

Comment:
This panics because of your code which you deliberately wrote. An if statement with three conditions with the OR relationship each of them...

I'm more interested in if your assumptions made based on SOMETHING and materialized by that strange if statement have any good fundamentals. In other words, I cannot believe the None here would cause exactly what you're saying, via the call of `sign_raw_transaction()`.

Reply:
**URGENCY STATUS: PLEASE REASSURE TOGETHER WITH UTKARSH THAT THIS IS A REALISTIC CONCERN.**
How did you come to these extra measures for the none values. Is that even possible in our code to happen? Why do we test on it during runtime?

Reply:
We have changed the wording of these comments and the panic as it was confusing.
New Panic message: "We don't want to fetch any values while signing"
This is a Panic message for us the developers, here we don't want to fetch any of these values while signing.
This was more a message to let other developers know what we were thinking.

Reply:
I think this is cool. I'm up for closing this conversation. Please do it yourself though.